DuckDuckGo/YoutubePlayer/DuckPlayerOverlayPixels.swift

```diff
@@ -16,84 +16,93 @@
-        let comparisonURL = url.forComparison()
-
-        // Only append the URL if it's different from the last entry in normalized form
-        navigationHistory.append(comparisonURL)
-
-        // DuckPlayer is in Ask Mode, there's navigation history, and last URL is a YouTube Watch Video
-        guard duckPlayerMode == .alwaysAsk,
-              navigationHistory.count > 1,
-              let currentURL = navigationHistory.last,
-              let previousURL = navigationHistory.dropLast().last,
-              previousURL.isYoutubeWatch else { return }
-
-        var isReload = false
-        // Check for a reload condition: when current videoID is the same as Previous
-        if let currentVideoID = currentURL.youtubeVideoParams?.videoID,
-           let previousVideoID = previousURL.youtubeVideoParams?.videoID,
-           !previousURL.isDuckPlayer, !currentURL.isDuckPlayer {
-            isReload = currentVideoID == previousVideoID
-        }
+    deinit {
+        removeObservers()
+    }
 
-        // Fire the reload pixel if this is a reload navigation
-        if isReload {
-            pixelFiring?.fire(GeneralPixel.duckPlayerYouTubeOverlayNavigationRefresh)
-        } else {
-            // Determine if it’s a back navigation by looking further back in history
-            let isBackNavigation = navigationHistory.count > 2 &&
-                                   navigationHistory[navigationHistory.count - 3].forComparison() == currentURL.forComparison()
-
-            // Fire the appropriate pixel based on navigation type
-            if isBackNavigation {
-                pixelFiring?.fire(GeneralPixel.duckPlayerYouTubeOverlayNavigationBack)
-            } else if previousURL.isYoutubeWatch && currentURL.isYoutube {
-                // Forward navigation within YouTube (including non-video URLs)
-                pixelFiring?.fire(GeneralPixel.duckPlayerYouTubeNavigationWithinYouTube)
-            } else if previousURL.isYoutubeWatch && !currentURL.isYoutube && !currentURL.isDuckPlayer {
-                // Navigation outside YouTube
-                pixelFiring?.fire(GeneralPixel.duckPlayerYouTubeOverlayNavigationOutsideYoutube)
-                navigationHistory.removeAll()
-            }
+    func fireNavigationPixelsIfNeeded(webView: WKWebView) {
+
+        guard let currentURL = webView.url else {
+            return
         }
+        
+        let backItemURL = webView.backForwardList.backItem?.url
 
-        // Truncation logic: Remove all URLs up to the last occurrence of the current URL in normalized form
-        if navigationHistory.count > 0 {
-            if let lastOccurrenceIndex = (0..<navigationHistory.count - 1).last(where: { navigationHistory[$0].forComparison() == comparisonURL }) {
-                navigationHistory = Array(navigationHistory.prefix(upTo: lastOccurrenceIndex + 1))
+        if lastVisitedURL != nil {
+            // Back navigation
+            if currentURL == backItemURL {
```

Comment:
This would fire "back" pixel also when you typed the back URL in the address bar but I think that for the purpose of measuring user behavior this is equivalent to the "back" behavior 👍

Reply:
yup, typing prev url is fine.  That also means going back